guillotina_gcloudstorage/storage.py

```diff
@@ -204,34 +204,47 @@ def should_clean(self, file):
         else:
             raise AttributeError('No valid uri')
 
-    async def _append(self, dm, data, offset):
-        if dm.size:
+    async def _append(self, dm, data, offset, is_last):
+        if dm.size is not None:
             size = dm.size
+        elif is_last:
+            size = len(data) + offset
```

Comment:
okay, I think I get it! This is clever and allows the person patching to not specify total size ever I guess--I'm not sure that's a bit thing.

Reply:
Yes, this is for when user doesn't specify size